### DISCUSSION TITLE
Questions about SCM HA setup

### DISCUSSION THREAD

**Discussion by shichuzhu** [apache/ozone#7299] [FAQ] [answered]
I'm trying to follow the guide to setup SCM with HA.

The [scm.client.address](https://github.com/apache/ozone/blob/ffe71985e63bc50c243db72e0fa61fd443dfd638/hadoop-hdds/common/src/main/resources/ozone-default.xml#L891-L893) property is tagged as required.

On the other hand, the [SCM HA guide](https://ozone.apache.org/docs/current/feature/scm-ha.html) has an example to setup 3 SCM nodes.

My questions are

1. What value I should put in the scm.client.address property? What is consuming this value? Is it the Datanode + OM using it? I assume it's the IP:port of the primordial node. And next question
2. What happens when the primordial SCM node fails? Will the system work properly by having all the consumers of scm.client.address auto fallback to the next SCM node? Is there anyway I can verify the current leader of SCM?
3. Is it possible for me to actively switch the SCM leader from the current primordial node to another node, even when no failure is happening?

**Comment by yazynin** [comment, marked as answer]
1 scm config looks like
`<property><name>ozone.scm.ratis.enable</name><value>true</value></property>
<property><name>ozone.scm.service.ids</name><value>cluster1</value></property>
<property><name>ozone.scm.nodes.cluster1</name><value>scm1,scm2,scm3</value></property>
<property><name>ozone.scm.address.cluster1.scm1</name><value>server1.com</value></property>
<property><name>ozone.scm.address.cluster1.scm2</name><value>server2.com</value></property>
<property><name>ozone.scm.address.cluster1.scm3</name><value>server3.com</value></property>
<property><name>ozone.scm.primordial.node.id</name><value>scm1</value></property>
`
2 yes clients connect to another node

3 ozone admin scm transfer

**Comment by mmartinez57** [comment]
> 2 Is there anyway I can verify the current leader of SCM?

`ozone admin scm roles`

**Comment by shichuzhu** [comment]
Thanks for the answers!
I followed the instructions and did NOT specify the `scm.client.address` parameter. The setup seems successful.
The takeaway is that the `scm.client.address` is NOT required in the SCM HA mode.
